src/mds/MDSRank.cc

```diff
@@ -1019,14 +1019,7 @@ void MDSRank::ProgressThread::shutdown()
 
 bool MDSRankDispatcher::ms_dispatch(const cref_t<Message> &m)
 {
-  if (m->get_source().is_mds()) {
-    const Message *msg = m.get();
-    const MMDSOp *op = dynamic_cast<const MMDSOp*>(msg);
-    if (!op)
-      dout(0) << typeid(*msg).name() << " is not an MMDSOp type" << dendl;
-    ceph_assert(op);
-  }
-  else if (m->get_source().is_client()) {
+  if (m->get_source().is_client()) {
```

Comment:
`get_source().is_client()` uses header data, I would suggest looking at the peer_type that you can get from connection, you can use `is_a_client()` from `Message.h` that I wrote as a helper function in https://github.com/ceph/ceph/pull/47080/commits/ae5ff477d21134afe6a017db828549c8c350b806

Reply:
I don't think this makes sense here. We won't handle the msgs from MDSs in this scope.

Reply:
Alright